src/impl_/pyclass.rs

```diff
@@ -813,7 +813,7 @@ slot_fragment_trait! {
         // By default `__ne__` will try `__eq__` and invert the result
         let slf: &PyAny = py.from_borrowed_ptr(slf);
         let other: &PyAny = py.from_borrowed_ptr(other);
-        slf.eq(other).map(|is_eq| PyBool::new(py, !is_eq).into_ptr())
+        slf.eq(other).map(|is_eq| PyBool::new_bound(py, !is_eq).as_ptr())
```

Comment:
Since the pointer points into a static, this should be fine, I think

Reply:
I think for reference counting sake we need to return an owned "reference" (in Python's C API terminology) from this method:

```suggestion
        slf.eq(other).map(|is_eq| PyBool::new_bound(py, !is_eq).to_owned().into_ptr())
```

Reply:
Yeah, this is what I was wondering. After looking at the Python C API docs again, this makes sense. `Py_INCREF` and `Py_DECREF` should handle this and compile to a noop in 3.12 and up where these are "immortal".

Reply:
I think it won't quite compile to nothing as the Rust optimizer won't know that bool is immortal, but it should still basically be a cheap operation that doesn't change the reference count.